pandas/tests/indexing/test_multiindex.py

```diff
@@ -158,12 +158,44 @@ def test_loc_getitem_series(self):
         result = x.loc[[1, 3]]
         tm.assert_series_equal(result, expected)
 
+        # GH15424
+        y1 = Series([1, 3], index=[1, 2])
+        result = x.loc[y1]
+        tm.assert_series_equal(result, expected)
+
         empty = Series(data=[], dtype=np.float64)
         expected = Series([], index=MultiIndex(
             levels=index.levels, labels=[[], []], dtype=np.float64))
         result = x.loc[empty]
         tm.assert_series_equal(result, expected)
 
+    def test_loc_getitem_dataframe(self):
+        # passing a dataframe as a key with a MultiIndex
+        index = MultiIndex.from_product([[1, 2, 3], ['A', 'B', 'C']])
+        x = Series(index=index, data=range(9), dtype=np.float64)
+        idx_keys = [(1, 'A'), (2, 'C'), (3, 'B')]
```

Comment:
can you also add tests for empty Series/DataFrame indexers (Series in test above)